providers/cstor/cstor-block-device-pool/test.yml

```diff
@@ -15,178 +15,231 @@
+                  line: '    - {{ item }}'
+                with_items: "{{ bd_names }}"
+
+              when: pool_type == "raidz"
+
+            - block:
+
+              - name: Getting the Unclaimed block-device from each node when pool type is RAIDZ2
+                shell: kubectl get blockdevice -n openebs -l kubernetes.io/hostname={{ item }} -o jsonpath='{.items[?(@.status.claimState=="Unclaimed")].metadata.name}' | tr " " "\n" | grep -v sparse | head -n 6
+                register: blockDevice
+                with_items: "{{ nodes.stdout_lines }}"
+
+              - name: Initialize an empty list to store block device names
+                set_fact:
+                  bd_names: []
+
+              - name: Store name of all block devices in the list
+                set_fact:
+                  bd_names: "{{ bd_names + item.stdout_lines }}"
+                with_items:
+                  - "{{ blockDevice.results }}"
+                
+              - name: Adding discovered block device into SPC spec
+                lineinfile:
+                  path: "./spc.yml"
+                  insertafter: 'blockDeviceList:'
+                  line: '    - {{ item }}'
+                with_items: "{{ bd_names }}" 
+              
+              when: pool_type == "raidz2"  
+
+            - set_fact:
+                device_count: "{{ blockDevice|length}}"  
+
+            - name: Replacing the pool name in SPC spec
+              replace:
+                path: ./spc.yml
+                regexp: "pool-name"
+                replace: "{{ pool_name }}"
+
+            - name: Replacing the maxPool count in SPC spec
+              replace:
+                path: ./spc.yml
+                regexp: "maxpool-count"
+                replace: "{{ maxpool_count }}"
+
+            - name: Replacing the storage class name in SPC spec
+              replace:
+                path: ./spc.yml
+                regexp: "sc-name"
+                replace: "{{ sc_name }}"
+
+            - name: Display spc.yml for verification 
+              debug: var=item
+              with_file:
+              - "spc.yml"
+
+            - name: Create cstor disk pool
+              shell: kubectl apply -f spc.yml
+              args:
+                executable: /bin/bash
+
+            - name: Verify if cStor Disk Pool are Running
+              shell: >
+                kubectl get pods -n {{ operator_ns }} -l openebs.io/storage-pool-claim={{ pool_name }}
+                --no-headers -o custom-columns=:status.phase
+              args:
+                executable: /bin/bash
+              register: pool_count
+              until: "((pool_count.stdout_lines|unique)|length) == 1 and 'Running' in pool_count.stdout"
+              retries: 30
+              delay: 10
+
+            - name: Obtain the maxPool clount from spc
+              shell: >
+                kubectl get spc {{ pool_name }} -o custom-columns=:.spec.maxPools --no-headers
```

Comment:
Can we get the maxPools from json path?

Reply:
This will not help in the manual pool provisioning case.
If it is a manual pool provisioning case we need to get pool count based on `device_count/disk_required_for_pool`